integration/healthcheck_test.go

```diff
@@ -67,16 +55,40 @@ func (s *HealthCheckSuite) TestSimpleConfiguration(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, 200)
 
-	healthReq, err := http.NewRequest("POST", "http://"+whoami1Host+"/health", bytes.NewBuffer([]byte("500")))
+	// Fix all whoami health to 500
+	whoamiHosts := []string{whoami1Host, whoami2Host}
+	for _, whoami := range whoamiHosts {
+		healthReq, err := http.NewRequest("POST", "http://"+whoami+"/health", bytes.NewBuffer([]byte("500")))
+		c.Assert(err, checker.IsNil)
+		_, err = client.Do(healthReq)
+		c.Assert(err, checker.IsNil)
+	}
+
+	utils.Sleep(3 * time.Second)
+
+	// Verify frontend health : 500
+	err = utils.TryRequest(req, 3*time.Second, utils.UntilStatusCodeIs(500))
 	c.Assert(err, checker.IsNil)
-	_, err = client.Do(healthReq)
+
+	// Change one whoami health to 200
+	healthReq1, err := http.NewRequest("POST", "http://"+whoami1Host+"/health", bytes.NewBuffer([]byte("200")))
+	c.Assert(err, checker.IsNil)
+	_, err = client.Do(healthReq1)
 	c.Assert(err, checker.IsNil)
 
-	time.Sleep(time.Second * 3)
+	// Verify frontend health : before
+	err = utils.TryRequest(req, 3*time.Second, utils.UntilStatusCodeIs(500))
+	c.Assert(err, checker.IsNil)
 
-	resp, err = client.Do(req)
+	utils.Sleep(3 * time.Second)
```

Comment:
I'm not sure about that one, at all. What are we testing here : that the health was `500` before (it's tested above) and that in a given period of time, it will be `200`. I don't see why we re-verify `500`, wait for 3s and then wait for `200`.